cpp/include/cudf/table/row_operators.cuh

```diff
@@ -345,6 +345,41 @@ class row_lexicographic_comparator {
     CUDF_EXPECTS(_lhs.num_columns() == _rhs.num_columns(), "Mismatched number of columns.");
   }
 
+  inline __device__ weak_ordering struct_compare(column_device_view const& lhs,
+                                                 column_device_view const& rhs,
+                                                 null_order null_precedence,
+                                                 size_type lhs_index,
+                                                 size_type rhs_index) const noexcept
+  {
+    if (has_nulls) {
+      bool const lhs_is_null{lhs.nullable() and lhs.is_null(lhs_index)};
+      bool const rhs_is_null{rhs.nullable() and rhs.is_null(rhs_index)};
+
+      if (lhs_is_null and rhs_is_null) {  // null <? null
+        return weak_ordering::EQUIVALENT;
+      } else if (lhs_is_null) {  // null <? x
+        return (null_precedence == null_order::BEFORE) ? weak_ordering::LESS
+                                                       : weak_ordering::GREATER;
+      } else if (rhs_is_null) {  // x <? null
+        return (null_precedence == null_order::AFTER) ? weak_ordering::LESS
+                                                      : weak_ordering::GREATER;
+      }
```

Comment:
Let's DRY this up and put it into a function:
```
auto null_compare(bool lhs_is_null, bool rhs_is_null, null_order){
     if (lhs_is_null and rhs_is_null) {  // null <? null
        return weak_ordering::EQUIVALENT;
      } else if (lhs_is_null) {  // null <? x
        return (null_precedence == null_order::BEFORE) ? weak_ordering::LESS
                                                       : weak_ordering::GREATER;
      } else if (rhs_is_null) {  // x <? null
        return (null_precedence == null_order::AFTER) ? weak_ordering::LESS
                                                      : weak_ordering::GREATER;
      }
}
```

Reply:
I thought of DRY. but there is no `else` case or default return value. `else` case is actually a value comparison.

Reply:
Looks like the else/default return is ` weak_ordering state{weak_ordering::EQUIVALENT};` from line 370?